auth/src/android/credential_android.cc

```diff
@@ -441,6 +457,58 @@ Credential EmailAuthProvider::GetCredential(const char* email,
   return cred;
 }
 
+//
+// PhoneAuthCredential methods
+//
+PhoneAuthCredential::PhoneAuthCredential() {}
+
+PhoneAuthCredential::PhoneAuthCredential(void* impl) {
+  impl_ = impl;
+  error_code_ = kAuthErrorNone;
+}
+
+PhoneAuthCredential::PhoneAuthCredential(const PhoneAuthCredential& rhs) {
+  if (rhs.impl_ != nullptr) {
+    JNIEnv* env = GetJniEnv();
+    jobject j_cred_ref = env->NewGlobalRef(static_cast<jobject>(rhs.impl_));
+    impl_ = static_cast<void*>(j_cred_ref);
+  } else {
+    impl_ = nullptr;
+  }
+  error_code_ = rhs.error_code_;
+  error_message_ = rhs.error_message_;
+}
```

Comment:
Similarly, this is pretty much the same implementation to `operator=()`, isn't it?
Why not just use `*this = rhs;`?

Reply:
Fixed!